include/mlir/IR/DialectImplementation.h

```diff
@@ -188,6 +188,12 @@ class DialectAsmParser {
   /// Parse a `}` token.
   virtual ParseResult parseRBrace() = 0;
 
+  /// Parse a '{' token if present
+  virtual ParseResult parseOptionalLBrace() = 0;
+
```

Comment:
Can you keep the same ordering as the .cpp file?

Reply:
You bet.